src/header/common/host.rs

```diff
@@ -97,6 +64,29 @@ impl HeaderFormat for Host {
     }
 }
 
+impl FromStr for Host {
+    type Err = ::Error;
+
+    fn from_str(s: &str) -> ::Result<Host> {
+        let (host_port, res) = domain_to_unicode(s);
```

Comment:
Nice! I had noticed that fixme and was going to file a separate issue, but you win!